File: sort/가장큰수.js
function solution(numbers) {
  let answer = "";
  numbers = numbers.map((item) => (item + "").repeat(3));
  numbers.sort().reverse();
  numbers = numbers.map((item) => item.slice(0, item.length / 3));
  answer = numbers.join("");
  //     큰 숫자는 e를 사용하여 표현되어 우리가 원하는 형식으로 표현되지 않는다.
  // answer = Number(numbers.join('')).toString();
  //     따라서 예외적으로 0000 같이 0으로만 구성된 문자열에 대해서 0으로 바꿔줄 필요가 있다.
  if (Number(answer) === 0) {
    return "0";
  }
  return answer;
}

File: sort/가장큰수.java
import java.util.Arrays;

class Solution {
    public String solution(int[] numbers) {
        // String answer = "";
        int n = numbers.length;
        String[] arr = new String[n];
        for (int i = 0; i < n; i++) {
            arr[i] = String.valueOf(numbers[i]);
        }

        // Arrays.sort(arr, new Comparator<String>() {
        //     @Override
        //     public int compare(String o1, String o2) {
        //         return ((o2 + o1).compareTo(o1 + o2));
        //     }
        // });
        Arrays.sort(arr, (s1, s2) -> (s2 + s1).compareTo(s1 + s2));
        
        if (arr[0].equals("0")) return "0";
        return String.join("", arr);
    }
}
